document_ai_warehouse/batch-ingestion-docai-wh/load_docs.py

```diff
@@ -0,0 +1,493 @@
+    mapping_dic = {
+        "display_name": display_name,
+        "property_definitions": [],
+        "document_is_folder": False,
+        "description": "Auto-generated using batch upload",
+    }
+
+    if options:
+        for name, value in names:
+            definition = {
+                "name": name,
+                "display_name": name,
+                "is_repeatable": False,
+                "is_filterable": True,
+                "is_searchable": True,
+                "is_metadata": True,
+                "is_required": False,
+            }
+
+            v_type = get_type(value)
+            if v_type:
+                definition[v_type] = {}
+            mapping_dic["property_definitions"].append(definition)
+
+    file_path = os.path.join(
+        os.path.dirname(__file__), "schema_files", f"{display_name}.json"
+    )
+    with open(file_path, "w") as f:
+        json.dump(mapping_dic, f, indent=2)
+
+    return file_path
+
+
+def document_exists(reference_id: str) -> bool:
+    reference_path = f"referenceId/{reference_id}"
+    try:
+        dw_utils.get_document(reference_path, CALLER_USER)
+        return True
+    except NotFound:
+        return False
+
+
+def delete_document(reference_id: str):
+    Logger.info(f"delete_document reference_id={reference_id}")
+    reference_path = f"referenceId/{reference_id}"
+    dw_utils.delete_document(document_id=reference_path, caller_user_id=CALLER_USER)
+
+
+def upload_document_gcs(
+    file_uri: str,
+    document_schema_id: str,
+    folder_id: str,
+    reference_id: str,
+    document_ai_output,
+    metadata_properties: List[contentwarehouse_v1.Property],
+):
+    create_document_response = dw_utils.create_document(
+        display_name=os.path.basename(file_uri),
+        mime_type="application/pdf",
+        document_schema_id=document_schema_id,
+        raw_document_path=file_uri,
+        docai_document=document_ai_output,
+        caller_user_id=CALLER_USER,
+        reference_id=reference_id,
+        metadata_properties=metadata_properties,
+    )
+
+    Logger.debug(
+        f"create_document_response={create_document_response}"
+    )  # Verify that the properties have been set correctly
+
+    if create_document_response is not None:
+        document_id = create_document_response.document.name.split("/")[-1]
+        document_id_list.append(document_id)
+
+        dw_utils.link_document_to_folder(
+            document_id=document_id,
+            folder_document_id=folder_id,
+            caller_user_id=CALLER_USER,
+        )
+        Logger.info(
+            f"Created document {file_uri} with reference_id={reference_id} inside folder_id={folder_id} and using schema_id={document_schema_id}"
+        )
+        return document_id
+
+
+def create_folder_schema(schema_path: str):
+    folder_schema = storage_utils.read_file(schema_path, mode="r")
+    display_name = json.loads(folder_schema).get("display_name")
```

Comment:
If this is using a ContentWarehouse data type, you can load it from a json file using

`ClassName.from_json()`